src/lib.rs

```diff
@@ -667,6 +669,33 @@ decl_module! {
 
             Ok(())
         }
+
+        #[weight = FunctionOf(
+            |args: (&T::AccountId, &u8, &[u8; 32], &Timepoint<T::BlockNumber>)| 500_000 * (*args.1 as u32),
+            DispatchClass::Normal,
+            true
+        )]
+		fn clear(origin, multi_account_id: T::AccountId, max_cancellations: u8, call_hash: [u8; 32], timepoint: Timepoint<T::BlockNumber>) -> DispatchResult {
+			let mut succeeded = 0;
+			let who = ensure_signed(origin)?;
+			let mut cancelled_multisigs = Vec::with_capacity(max_cancellations as usize);
+			let active_multisigs = <Multisigs<T>>::iter_prefix(&who);
+			for m_sig in active_multisigs.take(max_cancellations as usize){
+				ensure!(m_sig.when == timepoint, Error::<T>::WrongTimepoint);
+				ensure!(who == multi_account_id || who == m_sig.depositor, Error::<T>::NotOwner);
+
+				let _ = T::Currency::unreserve(&m_sig.depositor, m_sig.deposit);
+				<Multisigs<T>>::remove(&multi_account_id, call_hash);
```

Comment:
I think every `call_hash` is specific for each multi sig, so we cannot know that upfront. We need a way to remove all multisigs per account_id only.

Reply:
This might work:  https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageDoubleMap.html#tymethod.remove_prefix

Reply:
The `unreserve` calls for each of the multi-sigs will be done up to the `max_cancellation`, but all are `multisigs` for that `multi_account_id` is removed regardless of the value of `max_cancellation`. Will this be problem?

Reply:
Yes, only `max_cancellation` multisigs should be cancelled.